README.md

```diff
@@ -1 +1,122 @@
+function ChunkedBase.consume!(::ConsumeContext, payload::ParsedPayload)
+    chunking_ctx = payload.chunking_ctx
+    result_buffer = payload.results
+
+    # To demonstrate how double-buffering works, print the buffer id and the refill number
+    chunk_info = (chunking_ctx.id, chunking_ctx.buffer_refills[])
+    @info "Newlines in chunk (id:$chunk_info): $(result_buffer.newlines_in_chunk)"
+    return nothing
+end
+```
+
+Now that we defined all overloads and contexts we can define our user-facing function:
+
+```julia
+# Prints relative positions of `\n` in each ingested chunk of size `buffersize`,
+# using `nworkers` spawned tasks.
+function print_newlines(io, buffersize, nworkers)
+    lexer = Lexer(io, nothing, '\n')
+    parsing_ctx = ParsingContext()
+    consume_ctx = ConsumeContext()
+    chunking_ctx = ChunkingContext(buffersize, nworkers, 0, nothing)
+    # ChunkedBase.jl requires 2 result buffers per worker task, we'd get an error otherwise
+    result_buffers = [ResultBuffer(Int[]) for _ in 1:2nworkers]
+    # We need to fill the byte buffer and find the newlines in it before we hand
+    # the `consume_ctx` to the function
+    ChunkedBase.read_and_lex!(lexer, chunking_ctx)
+    parse_file_parallel(lexer, parsing_ctx, consume_ctx, chunking_ctx, result_buffers)
+    return nothing
+end
+```
+Let's run it on some data:
+```julia
+julia> io = IOBuffer((("x" ^ 4095) * "\n") ^ 64); # 256KiB
+julia> print_newlines(io, 64*1024, 4);
+# [ Info: Newlines in chunk (id:(1, 1)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(1, 1)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(1, 1)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(1, 1)): [57344, 61440, 65536]
+# [ Info: Newlines in chunk (id:(2, 1)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(2, 1)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(2, 1)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(2, 1)): [57344, 61440, 65536]
+# [ Info: Newlines in chunk (id:(1, 2)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(1, 2)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(1, 2)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(1, 2)): [57344, 61440, 65536]
+# [ Info: Newlines in chunk (id:(2, 2)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(2, 2)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(2, 2)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(2, 2)): [57344, 61440, 65536]
+```
+Behind the scenes, `ChunkedBase.jl` was using two 64KiB buffers, finding newlines in them, and splitting the found newlines among 4 tasks. We can see that each of the buffers (identified by the first number in the `id` tuple) was refilled two times (the refill number is the second element of the tuple).
```

Comment:
users have to set buffer size? can there be docs on how to set it (i.e. how to decide)?

Reply:
I menation this a bit in the docstring of ChunkingContext